src/send_kcidb.py

```diff
@@ -570,6 +573,14 @@ def _run(self, context):
                 }
             }
             self._send_revision(context['client'], revision)
+
+            # mark the node as processed, sent_kcidb field to True
+            # TBD: job nodes might have child nodes, mark them as processed as well
+            node['sent_kcidb'] = True
+            try:
+                self._api.node.update(node)
```

Comment:
I think the safe place to update the node is in `_send_revision` method after data validation is done with `kcidb.io.SCHEMA.is_valid(revision)`. As for some reason, if data is invalid, the service would still set the `kcidb_sent` flag to `True`.

Reply:
I might move there, but may data become valid later?
Because if i will not flag node as processed, it will keep trying to resend it (and fail, because data is invalid).
Maybe better we just report invalid data somewhere.

Reply:
Data becomes invalid if incorrect type of data is stored in node or there is some parsing issue while generating KCIDB-compatible data dictionary in the bridge service.
That's why the same data won't be valid again without any changes to service or node model.
Maybe we can have an error log for invalid data. Also, we can have a `dict` field in `Node` model for KCIDB related processing information with 2 flags i.e. `kcidb_sent` and `processed_by_kcidb_bridge`, and use both the fields for a submission retry.

Reply:
Then maybe i will change name to processed_by_kcidb_bridge (but it feels a bit long), it will be more correct such way.